### DISCUSSION TITLE
Plan for `rust-gpu` and `rust-cuda`

### DISCUSSION THREAD

**Discussion by JackWolfard** [Rust-GPU/rust-gpu#7] [Q&A] [answered]
In the [transition announcement](https://rust-gpu.github.io/blog/transition-announcement/), it was announced

> If instead you wish to stick to the NVIDIA ecosystem, stay tuned as the [Rust CUDA](https://github.com/rust-gpu/rust-cuda) project is in the process of being rebooted and possibly integrated with Rust GPU.
> ...
> [rust-cuda](https://github.com/Rust-GPU/Rust-CUDA): Compiles unmodified Rust code to [NVVM](https://docs.nvidia.com/cuda/nvvm-ir-spec/index.html) (CUDA) so that it can run on the GPU. Currently unmaintained but stay tuned!

I am interested in hearing more about this reboot effort and who might be the new maintainers. Is there development already happening in the open somewhere?

**Comment by JackWolfard** [comment]
@LegNeato @eddyb can you comment?

**Comment by LegNeato** [comment, marked as answer]
Sure! I have recently become a maintainer of `rust-cuda` and behind the scenes I have been talking to the previous maintainer as well as the person working on the rustc ptx backend about the best way forward for the project technically.

 `rust-cuda` in the past has looked at the compiler infra bits `rust-gpu` has and mirrored some of them. It might make sense to merge codebases where we can and have cuda vs vulkan be more like arm vs x86 in rustc (that is, separate arch/platform specific implementations sharing overall project and infrastructure).

We've been focused on this project's transition before working on rebooting `rust-cuda`, but in a couple weeks once we get a release out and hit our stride I'll start to focus on `rust-cuda`.

**Reply by rydb** [reply]
Question. Why CUDA?

**Reply by JackWolfard** [reply]
> Why CUDA?

My original question was b/c CUDA often has the most advanced GPU compute features of any language. If I wanted to make use of something like TensorCores, then they're directly programmable with CUDA, but in SPIR-V or OpenCL or another language, they're a little harder to access or are more limited in feature set.
